packages/ui/src/components/Schemas/TextSchema.tsx

```diff
@@ -27,14 +27,23 @@ const TextSchemaUI = (
 
   useEffect(() => {
     if (schema.dynamicFontSize && schema.data) {
-      calculateDynamicFontSize({ textSchema: schema, font, input: schema.data }).then(setDynamicFontSize);
+      calculateDynamicFontSize({
+        textSchema: schema,
+        font,
+        input: schema.data,
+        startingFontSize: dynamicFontSize,
+      }).then(setDynamicFontSize);
     } else {
       setDynamicFontSize(undefined);
     }
   }, [
-    schema,
```

Comment:
Listening on parent objects will result in this useEffect being triggered even if that schema has not changed.
The same is true of listening on `schema.dyanmicFontSize`. 

Removing these and only listening to the attributes explicitly ensures we only trigger when relevant changes are made